amr-wind/boundary_conditions/wall_models/MOSD.H

```diff
@@ -0,0 +1,52 @@
+#ifndef MOSD_H
+#define MOSD_H
+
+#include "AMReX_AmrCore.H"
+#include <AMReX.H>
+namespace amr_wind {
+struct MOSD
+{
+    /*
+     * A dynamic wall model that calculates the stress from wave to wind
+     * based on geometric information of the wave.
+     */
+
+    amrex::Real amplitude;
+    amrex::Real wavenumber;
+    amrex::Real omega;
+    amrex::Real time;
+
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE amrex::Real
+    get_dyn_tau(amrex::Real u_dx, amrex::Real v_dx, amrex::Real xc, amrex::Real unit_nor) const
+    {
+        amrex::Real tau_wave;
+
+        // Building the wave surface, gradients, wave velocities and unit normal
+        amrex::Real dx_eta_wave = -amplitude * wavenumber * std::sin(wavenumber * xc  - omega * time);
```

Comment:
probably all these can be `const` declared.

Reply:
shouldn't declare them const because at each time step the value changes for all these variables.

Reply:
If the variable does not change in the function then it should be const. It signals to the reader/compiler that they can read that variable and see the value and not expect it to change later in the function they are reading. Most variables are typically const. Some are not (like counters in while loops)